graql/language/update.feature

```diff
@@ -0,0 +1,178 @@
+      """
+    Then transaction commits
+    When session opens transaction of type: read
+    When get answers of graql match
+      """
+      match $x isa person, has name $n;
+      """
+    Then uniquely identify answer concepts
+      | x          | n               |
+      | key:ref:0  | value:name:Alex |
+      | key:ref:1  | value:name:Bob  |
+
+  Scenario: Deleting the last roleplayer of a relation means it cannot be updated
+    Given get answers of graql insert
+      """
+      insert
+      $x isa person, has name "Alex", has ref 0;
+      $y isa person, has name "Bob", has ref 1;
+      $r (friend: $x, friend:$y) isa friendship, has ref 0;
+      """
+    Given transaction commits
+    Given session opens transaction of type: write
+    When graql update; throws exception
+      """
+      match
+        $x isa person, has name "Alex";
+        $y isa person, has name "Bob";
+        $r (friend: $x) isa friendship;
+      delete
+        $r (friend: $x);
+      insert
+        $r (friend: $y);
+      """
+
+  Scenario: Roleplayer exchange
+    Given get answers of graql insert
+      """
+      insert
+      $x isa person, has name "Alex", has ref 0;
+      $y isa person, has name "Bob", has ref 1;
+      $r (parent: $x, child:$y) isa parenthood;
+      """
+    Given transaction commits
+    Given session opens transaction of type: write
+    When graql update
+      """
+      match
+        $r (parent: $x, child: $y) isa parenthood;
+      delete
+        $r isa parenthood;
+      insert
+        (parent: $y, child: $x) isa parenthood;
+      """
+
+  Scenario: Complex migration
+    Given get answers of graql insert
+      """
+      insert
+      $u isa person, has name "Alex", has ref 0;
+      $v isa person, has name "Bob", has ref 1;
+      $w isa person, has name "Charlie", has ref 2;
+      $x isa person, has name "Darius", has ref 3;
+      $y isa person, has name "Alex", has ref 4;
+      $z isa person, has name "Bob", has ref 5;
+      """
+    Given connection close all sessions
+    Given connection open schema session for database: grakn
+    Given session opens transaction of type: write
+    Given graql define
+      """
+      define
+      nameclass sub entity,
+        owns name @key,
+        plays naming:name;
+      naming sub relation,
+        relates named,
+        relates name;
+      person plays naming:named;
+      """
+    Given transaction commits
+    Given connection close all sessions
+    Given connection open data session for database: grakn
+    Given session opens transaction of type: write
+    When graql insert
+      """
+      match
+        $att isa name;
+      insert
+        $x isa nameclass, has $att;
+      """
+    When graql update
+      """
+      match
+        $p isa person, has name $n;
+        $nc isa nameclass, has name $n;
+      delete
+        $p has $n;
+      insert
+        (named: $p, name: $nc) isa naming;
+      """
+    Then transaction commits
```

Comment:
Isn't there a value to doing the match and checking the answers? Otherwise how do we know this was successful?